atom.js

```diff
@@ -526,8 +526,11 @@
 
 			// Set value for a key, or if `keyOrMap` is an object then set all the
 			// keys' corresponding values.
-			set: function (keyOrMap, value, validateFirst) {
+			set: function (keyOrMap, valueOrSetConfig, setConfig) {
 				var localConfig = config;
+				var isMap = isObject(keyOrMap);
```

Comment:
I'd go with sth like `var config = Object.assign({ validate: true }, setConfig)`, then three lines below can be removed